addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/handler/UniFiControllerThingHandler.java

```diff
@@ -230,12 +235,28 @@ private void cachePut(Map<String, UniFiClient> cache, UniFiClient client) {
         }
 
         // mgb: short circuit
-        if (client == null || BooleanUtils.isNotTrue(client.isWireless()) || !belongsToSite(client, site)) {
+        if (client == null || !belongsToSite(client, site)) {
             return null;
         }
 
         // mgb: instanceof check just for type / cast safety
-        return (client instanceof UniFiWirelessClient ? (UniFiWirelessClient) client : null);
+        return client;
+    }
+
+    public UniFiClient @Nullable [] setClientBlock(UniFiClient client) throws UniFiException {
```

Comment:
These methods should not be in the class. It's just used here as a pass through, so this class should not have that knowledge. Instead create a method here to return the controller. Then the client handler can get the controller and directly call those methods.

Reply:
Moved!